src/main/java/org/interledger/cryptoconditions/Condition.java

```diff
@@ -1,56 +1,57 @@
-   * entire string or binary encoded condition as the key - not just the fingerprint - as different
-   * conditions of different types may have the same fingerprint.
+   * <p>A fingerprint is a binary (aka "octet") string uniquely representing the condition with
+   * respect to other conditions of the same type. Implementations which index conditions MUST use
+   * the  entire string or binary encoded condition as the key - not just the fingerprint - as
+   * different conditions of different rsa may have the same fingerprint. The length and contents of
+   * the fingerprint are defined by the condition type. The fingerprint is a cryptographically
+   * secure hash of the data which defines the condition, such as a public key.</p>
    *
-   * The length and contents of the fingerprint are defined by the condition type. The fingerprint
-   * is a cryptographically secure hash of the data which defines the condition, such as a public
-   * key.
+   * @return A read-only byte array that contains the unique fingerprint of this condition.
    *
-   * @return the unique fingerprint of this condition
+   * @deprecated This method may be removed in future versions, so callers should use {@link
+   *     #getFingerprintBase64Url()} instead. Java 8 does not currently have the concept of an
+   *     immutable byte array, which means outside callers could manipulate the bytes returned from
+   *     this method and unintentionally (or purposefully) mutate the internal state of this
+   *     Condition.
    */
+  @Deprecated
   byte[] getFingerprint();
 
   /**
-   * The estimated "cost" of processing a fulfillment of this condition. For details of how to
-   * calculate this number see the Crypto-conditions specification.
+   * <p>A fingerprint is a binary (aka "octet") string uniquely representing the condition with
+   * respect to other conditions of the same type. Implementations which index conditions MUST use
+   * the entire string or binary encoded condition as the key - not just the fingerprint - as
+   * different conditions of different rsa may have the same fingerprint. The length and contents of
+   * the fingerprint are defined by the condition type. The fingerprint is a cryptographically
+   * secure hash of the data which defines the condition, such as a public key.</p>
    *
-   * @return the cost of validating the fulfillment of this condition
+   * @return A Base64Url-encoded {@link String} that contains the unique fingerprint of this
+   *     condition.
    */
-  long getCost();
+  String getFingerprintBase64Url();
 
   /**
-   * Get the condition encoded in ASN.1 DER binary encoding.
+   * The estimated "cost" of processing a fulfillment of this condition. For details of how to
+   * calculate this number see the Crypto-conditions specification.
    *
-   * @return the DER encoded condition
+   * @return the cost of validating the fulfillment of this condition
    */
-  byte[] getEncoded();
```

Comment:
Hmmm...interesting. Yes, I _would_ like to separate encoding from the object because it makes everything much cleaner. The fulfillments and conditions can stand alone, and if you want to encode them, you can delegate that to several different alternatives.

Is there are reason that the Condition should mirror the `java.security.Key`? If so, then perhaps we should extend from Key and make that relationship overt? As it stands now, I feel like a `Key` and a `Condition` should be two different things, and if you want to do something that looks like a `Key`, then do your own translation....

(That's my instinct anyway - happy to hear counter-points)